fvm/evm/handler/handler_test.go

```diff
@@ -0,0 +1,530 @@
+package handler_test
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	gethParams "github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/evm/emulator"
+	"github.com/onflow/flow-go/fvm/evm/emulator/database"
+	"github.com/onflow/flow-go/fvm/evm/handler"
+	"github.com/onflow/flow-go/fvm/evm/testutils"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// TODO add test for fatal errors
+
+func TestHandler_TransactionRun(t *testing.T) {
+	t.Parallel()
+
+	t.Run("test - transaction run (happy case)", func(t *testing.T) {
+		testutils.RunWithTestBackend(t, func(backend types.Backend) {
+			testutils.RunWithTestFlowEVMRootAddress(t, backend, func(rootAddr flow.Address) {
+				testutils.RunWithEOATestAccount(t, backend, rootAddr, func(eoa *testutils.EOATestAccount) {
+
+					bs, err := handler.NewBlockStore(backend, rootAddr)
+					require.NoError(t, err)
+
+					aa, err := handler.NewAddressAllocator(backend, rootAddr)
+					require.NoError(t, err)
```

Comment:
Maybe add a helper function in this test suite called something like:
```
func runHandlerTest(name string, t *testing.T, func test(backend, bs, aa) {
  testutils.RunWithTestBackend(t, func(backend types.Backend) {
    testutils.RunWithTestFlowEVMRootAddress(t, backend, func(rootAddr flow.Address) {
      testutils.RunWithEOATestAccount(t, backend, rootAddr, func(eoa *testutils.EOATestAccount) {
        bs, err := handler.NewBlockStore(backend, rootAddr)
        require.NoError(t, err)
        
        aa, err := handler.NewAddressAllocator(backend, rootAddr)
        require.NoError(t, err)
        test(backend, bs, aa)
      }    
    }
  }
})
```

Reply:
in my view this parts needs a major refactoring to use the interfaces etc, I'll leave it for another PR to clean this part and add more tests